### PR TITLE
Updates to MOI v0.9

### PR DESCRIPTION
**Pull request by blegat**
Also drop Julia v0.6, Julia v0.7 and transition from REQUIRE to Project.toml.
~Requires https://github.com/JuliaOpt/MathOptInterface.jl/pull/742, https://github.com/JuliaOpt/MathOptInterface.jl/pull/743, https://github.com/JuliaOpt/MathOptInterface.jl/pull/744, https://github.com/JuliaOpt/MathOptInterface.jl/pull/745, https://github.com/JuliaOpt/MathOptInterface.jl/pull/746, https://github.com/JuliaOpt/MathOptInterface.jl/pull/747~

### CODE REVIEW

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -231,28 +251,57 @@ function MOI.optimize!(instance::Optimizer)
...
+    flag = optimizer.sol.ret_val
+    if flag == OPTIMIZE_NOT_CALLED
+        return "Optimize not called"
+    elseif flag == ECOS_OPTIMAL
+        return "Problem solved to optimality"
+    elseif flag == ECOS_PINF
+        return "Found certificate of primal infeasibility"
+    elseif flag == ECOS_DINF
+        return "Found certificate of dual infeasibility"
+    elseif flag == ECOS_INACC_OFFSET
```
Comment (mlubin) [inline src/MOI_wrapper.jl]: This is an incorrect interpretation of the flags. `ECOS_INACC_OFFSET` is a value that can be added to statuses to change the meaning. See https://github.com/JuliaOpt/ECOS.jl/blob/dbcf2e6ec707d6e3034bb4061e94f64d47cc29f1/src/MOI_wrapper.jl#L271.

File: .travis.yml [hunk 2]
```diff
@@ -3,11 +3,11 @@ os:
...
   - osx
 julia:
-  - 0.6
-  - 0.7
   - 1.0
+  - 1.1
 notifications:
   email: false
+before_script:
+  - julia -e 'using Pkg; Pkg.add(PackageSpec(name="MathOptInterface", rev="master"))'
```
Comment (mlubin) [inline .travis.yml]: Add a TODO to remove this before merging?

File: Project.toml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name = "ECOS"
+uuid = "e2685f51-7e38-5353-a97d-a921fd2c8199"
+repo = "https://github.com/JuliaOpt/ECOS.jl.git"
+version = "0.9.4"
```
Comment (mlubin) [inline Project.toml]: We ought to be bumping the minor version number when we drop support for julia releases.

File: src/MOI_wrapper.jl [hunk 4]
```diff
@@ -58,14 +60,32 @@ mutable struct Optimizer <: MOI.AbstractOptimizer
...
     end
 end
 
 MOI.get(::Optimizer, ::MOI.SolverName) = "ECOS"
 
+function MOI.set(optimizer::Optimizer, param::MOI.RawParameter, value)
+    optimizer.options[param.name] = value
+end
+function MOI.get(optimizer::Optimizer, param::MOI.RawParameter)
+    return optimizer.options[param.name]
```
Comment (mlubin) [inline src/MOI_wrapper.jl]: TODO: This gives a poor error message if the name of the parameter is invalid.

File: src/MOI_wrapper.jl [hunk 5]
```diff
@@ -231,28 +251,61 @@ function MOI.optimize!(instance::Optimizer)
...
     instance.data = nothing # Allows GC to free instance.data before A is loaded to ECOS
+    options = instance.options
+    if instance.silent
+        options = copy(options)
+        options[:verbose] = false
+    end
     ecos_prob_ptr = ECOS.setup(n, m, cone.f, cone.l, length(cone.qa), cone.qa,
-                               cone.ep, G, A, c, h, b; instance.options...)
+                               cone.ep, G, A, c, h, b; options...)
+    start_time = time()
```
Comment (mlubin) [inline src/MOI_wrapper.jl]: Nit: If setup is required for the solve, shouldn't it be included in the solve time?

File: src/MOI_wrapper.jl (hunk 4 above)
Comment (blegat) [reply src/MOI_wrapper.jl]: We are currently lacking a convention for that. Even `UniversalFallback` do not have a nice error:
https://github.com/JuliaOpt/MathOptInterface.jl/blob/52d166f91614d5d522bf055cac232c7c5ba3ea96/src/Utilities/universalfallback.jl#L104-L105
Should it be a `AttributeNotSetError` ? Or just `nothing` ?

File: src/MOI_wrapper.jl (hunk 5 above)
Comment (blegat) [reply src/MOI_wrapper.jl]: Makes sense, setup time is kind of solver specific

File: .travis.yml (hunk 2 above)
Comment (blegat) [reply .travis.yml]: When MOI is released, I remove this to lines and push to trigger the new build. I'll leave this comment unresolved as a reminder to be sure

**Pull request merged**